src/compare-tree/query.js

```diff
@@ -0,0 +1,37 @@
+import Emitter from '../emitter'
+
+import {
+  get,
+  inherit
+} from '../utils'
+
+import {
+  getKeyPaths
+} from './key-path'
+
+export default function Query (id, keys) {
+  Emitter.call(this)
+
+  this.revision = -1
+  this.id = id
+  this.keys = getKeyPaths(keys)
+}
+
+inherit(Query, Emitter, {
+  extract (state) {
+    let len = this.keys.length
+    let values = Array(len)
+
+    for (var i = 0; i < len; i++) {
+      values[i] = get(state, this.keys[i])
+    }
```

Comment:
i see a lot of array iteration like this in this PR. is the use of `.map` not ideal? seems like this could be written down to
```js
return this.keys.map(function (key) {
  get(state, key)
})
```